This patch fixes a bug that kernel will crash on boot.

Original patch description:

From 2a5fee2047e9c973c1d47481b7f03fe2dff515b8 Mon Sep 17 00:00:00 2001
From: zhangshuangshuang <zhangshuangshuang@ict.ac.cn>
Date: Fri, 17 Jun 2016 21:14:36 +0800
Subject: Loongson3: Adjust no-llsc option for different GCC
 version

Loongson3 GCC option "-mnofix-ls3a-llsc" has an alias
"-mno-fix-loongson3-llsc" on lemote platform, this patch
pass the rightful option name by automatic testing.

Change-Id: If02f072104c094068cdf79fad8a8ce05b66233d5
Signed-off-by: Zhang Shuangshuang <zhangshuangshuang@ict.ac.cn>

diff --git a/a/arch/mips/loongson64/Platform b/b/arch/mips/loongson64/Platform
index 0fce460..171cc16 100644
--- a/a/arch/mips/loongson64/Platform
+++ b/b/arch/mips/loongson64/Platform
@@ -22,7 +22,15 @@ ifdef CONFIG_CPU_LOONGSON2F_WORKAROUNDS
   endif
 endif

-cflags-$(CONFIG_CPU_LOONGSON3)	+= -Wa,--trap
+ifdef CONFIG_CPU_LOONGSON3
+  cflags-y += -Wa$(comma)--trap
+  ifeq ($(call cc-option-yn, -Wa$(comma)-mnofix-ls3a-llsc), y)
+    cflags-y += -Wa$(comma)-mnofix-ls3a-llsc
+  endif
+  ifeq ($(call cc-option-yn, -Wa$(comma)-mno-fix-loongson3-llsc), y)
+    cflags-y += -Wa$(comma)-mno-fix-loongson3-llsc
+  endif
+endif
 #
 # binutils from v2.25 on and gcc starting from v4.9.0 treat -march=loongson3a
 # as MIPS64 R2; older versions as just R1.  This leaves the possibility open
